src/GameClient.ts

```diff
@@ -45,7 +52,11 @@ export class GameClient extends Client {
                 return super.open({
                     authToken: options.authToken,
                     url: endpoints[0].address,
-                    extraHeaders: {
+                    extraHeaders:
+                    (options.sharecode !== undefined) ? {
```

Comment:
Sorry for the delay, ok i understand. 

The issue here is i don't like the repetition of `extraHeaders`.

how about we construct an object and then add attributes to it.

```
const extraHeaders = {
'X-Interactive-Version': options.versionId,
};
if(options.sharecode) {
extraHeaders['X-Interactive-Sharecode'] = options.sharecode;
}
```

@JohnyCilohokla